src/index.ts

```diff
@@ -93,38 +216,204 @@ export class ConsentControl {
      * @param {ITextResources} textResources the text want to be displayed
      */
     public setTextResources(textResources: ITextResources): void {
-        if (textResources.bannerMessageHtml) {
-            this.textResources.bannerMessageHtml = textResources.bannerMessageHtml;
+        for (let key of Object.keys(this.textResources)) {
+            let typeKey = <keyof ITextResources> key;
+
+            if (textResources[typeKey]) {
+                this.textResources[typeKey] = textResources[typeKey];
+            }
+        }
+    }
+
+    /**
+     * Set themes in the banner and preferences dialog
+     * If a theme is not included, the default one will be used
+     * 
+     * @param {IThemes} themes themes that will be used in the banner and preferences dialog
+     */
+    public setThemes(themes?: IThemes): void {
+        this.themes.light = this.defaultLightTheme;
+        if (themes?.light) {
+            this.setCertainTheme(themes.light, this.themes.light);
+        }
+
+        this.themes.dark = this.defaultDarkTheme;
+        if (themes?.dark) {
+            this.setCertainTheme(themes.dark, this.themes.dark);
         }
-        if (textResources.acceptAllLabel) {
-            this.textResources.acceptAllLabel = textResources.acceptAllLabel;
+
+        this.themes["high-contrast"] = this.defaultHighContrast;
+        if (themes?.["high-contrast"]) {
+            this.setCertainTheme(themes["high-contrast"], this.themes["high-contrast"]);
+        }
+    }
+
+    /**
+     * Use the passed theme to set the theme property
+     * 
+     * @param {ITheme} theme the passed theme that we want to display
+     * @param {ITheme} currentTheme the theme property that we want to set
+     */
+    public setCertainTheme(theme: ITheme, currentTheme: ITheme): void {
+        for (let key of Object.keys(currentTheme)) {
```

Comment:
`for of` is not supported by IE, which is important for us: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of

Two ways of solving that:
1. Update tsconfig.json to target lower version of Ecmascript. https://www.typescriptlang.org/docs/handbook/iterators-and-generators.html#code-generation
2. Rewrite without for...of

And please check that our solution works in IE